### PR TITLE
Create cli_services unittest

### PR DESCRIPTION
**Pull request by jasonborg**
Create cli_services unittest

### CODE REVIEW

File: snapshot_dbg_cli/cli_services.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 from snapshot_dbg_cli.firebase_management_rest_service import FirebaseManagementRestService
 from snapshot_dbg_cli.firebase_rtdb_rest_service import FirebaseRtdbRestService
 from snapshot_dbg_cli.gcloud_cli_service import GcloudCliService
+#import snapshot_dbg_cli.gcloud_cli_service
```
Comment (mctavish) [inline snapshot_dbg_cli/cli_services.py:25]: ?

File: snapshot_dbg_cli_tests/test_cli_services.py [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+    project_data['resources']['realtimeDatabaseInstance'] = defaut_rtdb_name
+
+  return FirebaseProject(project_data)
+
+
+class CliServicesTests(unittest.TestCase):
+  """ Contains the unit tests for the CliServices class.
+  """
+
+  def setUp(self):
```
Comment (mctavish) [inline snapshot_dbg_cli_tests/test_cli_services.py:48]: A quick comment indicating what's going on with the mocks below would be nice.

File: snapshot_dbg_cli_tests/test_cli_services.py [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+    self.gcloud_service_mock.config_get_account = MagicMock(
+        return_value='foo@bar.com')
+    self.gcloud_service_mock.config_get_project = MagicMock(
+        return_value='cli-test-project')
+    self.gcloud_service_mock.get_access_token = MagicMock(
+        return_value='cli-test-access-token')
+
+    expected_is_debug_enabled = False
+
+    cli_services = CliServices(args)
```
Comment (mctavish) [inline snapshot_dbg_cli_tests/test_cli_services.py:119]: Just another request for a quick comment.  It's clear from this test that creating a CliServices object causes a bunch of stuff to happen, but something more succinct than the assertions below would be handy.  (eg. gets project and auth info from gcloud)

File: snapshot_dbg_cli_tests/test_cli_services.py [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+    self.assertEqual('foo@bar.com', cli_services.account)
+    self.assertEqual('cli-test-project', cli_services.project_id)
+    self.assertEqual('cli-test-access-token', cli_services.access_token)
+    self.assertEqual(self.http_service_mock, cli_services.http_service)
+    self.assertEqual(self.permissions_service_mock,
+                     cli_services.permissions_service)
+    self.assertEqual(self.firebase_management_service_mock,
+                     cli_services.firebase_management_service)
+
+    self.data_formatter_class_mock.assert_called_once()
```
Comment (mctavish) [inline snapshot_dbg_cli_tests/test_cli_services.py:131]: I'm not sure how I feel about this one.  Is it just making sure that a constructor was called?

File: snapshot_dbg_cli_tests/test_cli_services.py [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+        self.user_output_class_mock.assert_called_once_with(
+            expected_is_debug_enabled, self.data_formatter_mock)
+
+  def test_get_snapshot_debugger_rtdb_service_url_provided(self):
+    args = argparse.Namespace()
+
+    service = CliServices(args).get_snapshot_debugger_rtdb_service(
+        database_url='https://fake-url.firebaseio.com')
+
+    self.assertEqual(service, self.snapshot_debugger_rtdb_service_mock)
```
Comment (mctavish) [inline snapshot_dbg_cli_tests/test_cli_services.py:185]: Is this making sure that the mocking worked or is there something more nuanced at play?

File: snapshot_dbg_cli_tests/test_cli_services.py [hunk 6]
```diff
@@ -0,0 +1,540 @@
...
+def build_firebase_project(defaut_rtdb_name=None):
+  project_data = {'resources': {}}
+
+  if defaut_rtdb_name is not None:
+    project_data['resources']['realtimeDatabaseInstance'] = defaut_rtdb_name
+
+  return FirebaseProject(project_data)
+
+
+class CliServicesTests(unittest.TestCase):
```
Comment (mctavish) [inline snapshot_dbg_cli_tests/test_cli_services.py:44]: In general, with this set of tests, I'm unclear on how many of the assertions are testing the implementation vs the functionality, nor where that distinction is important.  Happy to discuss.

File: snapshot_dbg_cli/cli_services.py (hunk 1 above)
Comment (jasonborg) [reply snapshot_dbg_cli/cli_services.py]: Thanks, done.

File: snapshot_dbg_cli_tests/test_cli_services.py (hunk 2 above)
Comment (jasonborg) [reply snapshot_dbg_cli_tests/test_cli_services.py:48]: Done. PTAL

File: snapshot_dbg_cli_tests/test_cli_services.py (hunk 3 above)
Comment (jasonborg) [reply snapshot_dbg_cli_tests/test_cli_services.py:155]: Done, PTAL.

File: snapshot_dbg_cli_tests/test_cli_services.py (hunk 6 above)
Comment (jasonborg) [reply snapshot_dbg_cli_tests/test_cli_services.py:44]: I've made some changes and added some comments. PTAL.

**Review by mctavish — APPROVED** [review]

File: snapshot_dbg_cli_tests/test_cli_services.py (hunk 4 above)
Comment (mctavish) [reply snapshot_dbg_cli_tests/test_cli_services.py]: Comments make it clear now.

**Pull request merged**